### ISSUE TITLE
file.comment always report changes in test=True mode

### ISSUE THREAD

**Issue by Talkless** [saltstack/salt#37939]
### Description of Issue/Question
`file.comment` prints `File <some_file> is set to be updated` on every test=True run, although there is actually nothing to comment any more.

### Setup
State file `comment_test.sls`:
```
#disable default PostgreSQL configuration in munin-node conf file:

munin_node_postgresql_plugin_config_comment_a:
    file.comment:
        - name: /tmp/munin-node
        - regex: |
            \[postgres_\*\]


munin_node_postgresql_plugin_config_comment_b:
    file.comment:
        - name: /tmp/munin-node
        - regex: |
            user postgres


munin_node_postgresql_plugin_config_comment_c:
    file.comment:
        - name: /tmp/munin-node
        - regex: |
            env\.PGPORT.*


munin_node_postgresql_plugin_config_comment_d:
    file.comment:
        - name: /tmp/munin-node
        - regex: |
            env\.PGUSER.*
```

And let's say we have file `/tmp/munin-node` with this content:
```
[postgres_*]
user postgres
env.PGUSER postgres
env.PGPORT 5432
```
### Steps to Reproduce Issue

- Call `state.apply comment_test` to apply comments.
- Check if lines are actually commented (they should be).
- Call `state.apply comment_test test=True ` for test run. Prints `File /tmp/munin-node is set to be updated` (that's **NOT OK**).
- Call `state.apply comment_test` to try apply comments again, but nothing actually happens with `Result: Clean` (and that's **OK**).

### Versions Report
(same versions and systems for master and minion)
```
Salt Version:
           Salt: 2016.3.4
 
Dependency Versions:
           cffi: 0.8.6
       cherrypy: 3.5.0
       dateutil: 2.2
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.0
   msgpack-pure: Not Installed
 msgpack-python: 0.4.2
   mysql-python: 1.2.3
      pycparser: 2.10
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.9 (default, Jun 29 2016, 13:08:31)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.4.0
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: debian 8.6 
        machine: x86_64
        release: 3.16.0-4-amd64
         system: Linux
        version: debian 8.6 
 ```

**Comment by Ch3LL** [comment]
@Talkless looks like i'm able to replicate this.

Here is a docker container to quickly replicate this:

1. `docker run -it -v /home/ch3ll/git/salt:/testing/ ch3ll/issues:37939 salt-call --local state.apply test test=True` (where /home/ch3ll/git/salt is a local cloned git repo)

We need to improve the test run reporting here thanks

**Comment by rallytime** [comment]
@Talkless Please give @38039 a try. That should fix this up for you. After playing around with the previously commented lines, we now get more useful test=True restults:
```
# salt-call --local state.apply test test=True
local:
----------
          ID: munin_node_postgresql_plugin_config_comment_a
    Function: file.comment
        Name: /tmp/munin-node
      Result: True
     Comment: Pattern already commented
     Started: 23:15:04.821283
    Duration: 7.254 ms
     Changes:
----------
          ID: munin_node_postgresql_plugin_config_comment_b
    Function: file.comment
        Name: /tmp/munin-node
      Result: None
     Comment: File /tmp/munin-node is set to be updated
     Started: 23:15:04.828663
    Duration: 1.63 ms
     Changes:
----------
          ID: munin_node_postgresql_plugin_config_comment_c
    Function: file.comment
        Name: /tmp/munin-node
      Result: None
     Comment: File /tmp/munin-node is set to be updated
     Started: 23:15:04.830402
    Duration: 2.256 ms
     Changes:
----------
          ID: munin_node_postgresql_plugin_config_comment_d
    Function: file.comment
        Name: /tmp/munin-node
      Result: True
     Comment: Pattern already commented
     Started: 23:15:04.832832
    Duration: 2.647 ms
     Changes:

Summary for local
------------
Succeeded: 4 (unchanged=2)
Failed:    0
------------
Total states run:     4
Total run time:  13.787 ms
```
That state run was from a file that I had uncommented like so:
```
# cat /tmp/munin-node
#[postgres_*]
user postgres
#env.PGUSER postgres
env.PGPORT 5432
```

@Ch3LL Thanks again for the awesome bug set up. It makes fixing things like this so much easier.

**Comment by Talkless** [comment]
@rallytime Thanks! It works as expected.

Will it be available for 2016.3.5?

**Comment by rallytime** [comment]
@Talkless Yeah, it will be in 2016.3.5. Thanks!
